src/main/java/com/productdock/library/search/book/BookService.java

```diff
@@ -21,8 +22,8 @@ public record BookService(BookDocumentRepository bookDocumentRepository,
                           RentalStateRecordMapper recordDocumentMapper) {
 
 
-    public SearchBooksResponse getBooks(Optional<List<String>> topics, int page) {
-        var hits = searchQueryExecutor.execute(topics, page);
+    public SearchBooksResponse getBooks(Optional<List<String>> topics, boolean recommended, int page) {
```

Comment:
Let's create class (ie SearchFilters) that will encapsulate search parameters. That way, method would have only 1 parameter, and it's better for the future when search by title is introduced.